addons/l10n_id_efaktur/views/account_move_views.xml

```diff
@@ -6,16 +6,14 @@
             <field name="model">account.move</field>
             <field name="inherit_id" ref="account.view_move_form"/>
             <field name="arch" type="xml">
-                <field name="partner_shipping_id" position="before">
-                    <field name="l10n_id_need_kode_transaksi" invisible="1"/>
-                    <field name="l10n_id_kode_transaksi" invisible="country_code != 'ID' or not l10n_id_need_kode_transaksi" readonly="state in ['cancel', 'posted']" required="l10n_id_need_kode_transaksi"/>
-                    <field name="l10n_id_efaktur_range" invisible="country_code != 'ID' or state != 'draft' or l10n_id_available_range_count &lt; 2 or not l10n_id_need_kode_transaksi" options="{'no_create': True}"/>
-                </field>
                 <button name="button_draft" position="after">
                     <button name="reset_efaktur" string="Reset E-Faktur" type="object" invisible="country_code != 'ID' or not l10n_id_tax_number or state != 'cancel' or l10n_id_efaktur_document"/>
                 </button>
                 <xpath expr=".//group[@id='other_tab_group']" position="inside">
                     <group string="Electronic Tax" invisible="country_code != 'ID'">
+                        <field name="l10n_id_need_kode_transaksi" invisible="1"/>
+                        <field name="l10n_id_kode_transaksi" invisible="country_code != 'ID' or not l10n_id_show_kode_transaksi" readonly="l10n_id_tax_number" required="l10n_id_need_kode_transaksi"/>
```

Comment:
```suggestion
                        <field name="l10n_id_kode_transaksi" invisible="not l10n_id_show_kode_transaksi" readonly="l10n_id_tax_number" required="l10n_id_need_kode_transaksi"/>
```

Where does it generate the number if you can modify it after the move was posted when you don't have a number?

Reply:
It won't generate it after posting in any case yes.
The idea of this fix is not to allow that but to avoid being stuck when:
- An invoice is auto generated and posted for a new customer
- The salesperson/accountat/... then add the vat number of the customer
- They now want to get a number => can't after posting so they want to reset to draft => but cannot because now, the code is required, not editable and empty.

The idea is to unblock the situation so they can set the code => reset to draft => post => re-reconcile.
It's not ideal but a fix in stable and what they were doing before.

I'm discussing with our Indo team to see if we can instead generate the number when we export the efaktur file, and avoid all of these issues but it would be a imp in master

Thanks for the review :)